packages/webdriverio/src/types.ts

```diff
@@ -18,46 +18,11 @@ type $ElementCommands = typeof ElementCommands
 type ElementQueryCommands = '$' | 'custom$' | 'shadow$' | 'react$'
 type ElementsQueryCommands = '$$' | 'custom$$' | 'shadow$$' | 'react$$'
 type ChainablePrototype = {
-    [K in ElementQueryCommands]: (...args: Parameters<$ElementCommands[K]>) => ChainablePromiseElement<ThenArg<ReturnType<$ElementCommands[K]>>>
+    [K in ElementQueryCommands]: (...args: Parameters<$ElementCommands[K]>) => Awaited<ThenArg<ReturnType<$ElementCommands[K]>>>
 } & {
-    [K in ElementsQueryCommands]: (...args: Parameters<$ElementCommands[K]>) => ChainablePromiseArray<ThenArg<ReturnType<$ElementCommands[K]>>>
+    [K in ElementsQueryCommands]: (...args: Parameters<$ElementCommands[K]>) => Awaited<ThenArg<ReturnType<$ElementCommands[K]>>>
 }
 
-type AsyncElementProto = {
-    [K in keyof Omit<$ElementCommands, keyof ChainablePrototype>]: OmitThisParameter<$ElementCommands[K]>
-} & ChainablePrototype
-
-interface ChainablePromiseBaseElement {
-    /**
-     * WebDriver element reference
-     */
-    elementId: Promise<string>
-    /**
-     * parent of the element if fetched via `$(parent).$(child)`
-     */
-    parent: Promise<WebdriverIO.Element | WebdriverIO.Browser | WebdriverIO.MultiRemoteBrowser>
-    /**
-     * selector used to fetch this element, can be
-     * - undefined if element was created via `$({ 'element-6066-11e4-a52e-4f735466cecf': 'ELEMENT-1' })`
-     * - a string if `findElement` was used and a reference was found
-     * - or a functin if element was found via e.g. `$(() => document.body)`
-     */
-    selector: Promise<Selector>
-    /**
-     * Error message in case element fetch was not successful
-     */
-    error?: Promise<Error>
-    /**
-     * index of the element if fetched with `$$`
-     */
-    index?: Promise<number>
-}
-export interface ChainablePromiseElement<T> extends
-    ChainablePromiseBaseElement,
-    AsyncElementProto,
-    Promise<T>,
```

Comment:
I was thinking of only removing this. I am not sure if removing `ChainablePromiseElement` would be a good idea.

Reply:
I find that my colleagues have no clue what `ChainablePromiseElement` is and just `await` it and I cannot blame them. I think having `WebdriverIO.Element` as type makes much more sense, what do you think?

Reply:
I think the noticeable difference is that:

```ts
const chainablePromiseElement = $('elem')
console.log(chainablePromiseElement.selector) // outputs: "Promise<string>"
```
whereas:

```ts
const wdioElement = await $('elem')
console.log(wdioElement.selector) // outputs: "elem"
```

So I am not sure if we can merge this into one single type since the chainable element is not resolved and accessing properties requires to use `await`. I can see that `ChainablePromiseElement` is confusing and I am in favor to consolidate this. Any suggestion how we can overcome the type difference?

Reply:
Interesting, does this mean that you need to `await` the `$` command when accessing any properties but not when you want to execute an action or chain it?

Reply:
Correct. You can do `$('elem').$('elem').$('elem').$('elem')` as our [proxy command wrapper](https://github.com/webdriverio/webdriverio/blob/58eb3c2914a49d2918c4a3ebe568b33280f6db94/packages/wdio-utils/src/shim.ts#L85-L262) allows to chain these promises.

Reply:
I believe you can extend the type to check if it has the elementId and if not you can assign it a Promise like type. I knowledge of Typescript is not where I want it to be so I might get a course on it to learn about more complex types 👍